source/emitCustomAttributes.ts

```diff
@@ -0,0 +1,80 @@
+export function emitCustomAttributes(customAttributes: CustomAttributeConfiguration[]) {
+    return `
+      export function getAttributeConfigurations() {
+        return [
+          ${customAttributes.map(
+        (x) => `{
+              name: "${x.key}",
+              type: ${JSON.stringify(x.type?.name)},
+              label: "${x.label}",
+              entityType: "${x.entity_type}",
+              default: ${JSON.stringify(x.default)},
+              objectType: ${JSON.stringify(x.object_type?.name)},
+              isHierarchical: ${x.is_hierarchical}
+            }`
+    )}
+        ] as const;
+      }
+      
+      export type RuntimeCustomAttributeConfiguration = ReturnType<typeof getAttributeConfigurations>[number];
+      export type RuntimeCustomAttributeConfigurationName = RuntimeCustomAttributeConfiguration["name"];
+      export type RuntimeCustomAttributeConfigurationLabel = RuntimeCustomAttributeConfiguration["label"];
+  
+      ${customAttributes
+        .map(x => `type ${getCustomAttributeTypeNameFromAttribute(x)} = Omit<ContextCustomAttributeValue, 'key' | 'value'> & {
```

Comment:
I'd prefer using a `BaseCustomAttributeValue` that doesn't have `key` and `value`, and extending from that. Preferably then that's what's used to create the untyped `ContextCustomAttributeValue` type, but if it's easier you can also create that separately as `Omit<ContextCustomAttributeValue, 'key' | 'value'>` and only use it for the typed custom attributes.

Reply:
This is now resolved.